src/modules/report/plain.py

```diff
@@ -14,15 +14,21 @@ class PlainReporter(object):
     def get_report(self):
         """generates report tables"""
         output = ""
+        services_lock.acquire()
```

Comment:
I'd definitely prefer to make this simpler by just grabbing the locks briefly to get the lengths. That way you can have a one-to-one relationship between acquire() and release() which is easier to read. 

Fortunately, you only ever need to hold one lock at a time (which is in general a very good thing - having to hold more than one lock often ends up creating deadlocks). However, at the moment you have to read the code quite carefully to see that's the case. If you use local variables to store len(services) and len(vulnerabilities) it will be really obvious that you only hold one lock at a time.